src/Hl7.Fhir.ElementModel/Language/TypeSpecifier.cs

```diff
@@ -115,22 +117,24 @@ public static TypeSpecifier ForNativeType(Type dotNetType)
             else if (t<int>() || t<short>() || t<ushort>() || t<uint>())
```

Comment:
No, you cannot. This is matching types against types, not pattern matching an instance against a type.  And since typeof(int) is not considered a constant, you cannot use it in a switch.

Reply:
(that's why I invented that super-short `t<T>()` - to make it at least look like a case.